csrc/preseg_passes/reorder_sharded_axis.cpp

```diff
@@ -72,6 +72,11 @@ void ReorderShardedAxisPass::runPass(Fusion* fusion) {
       new_output->axis(sharding_axis)->parallelize(ParallelType::Serial);
       output_permute->setDeviceMesh(output->getDeviceMesh());
       new_output->setDeviceMesh(output->getDeviceMesh());
+
+      // Set allocation domain for comm in/out
+      input_permute->setAllocationDomain(input_permute->getLoopDomain(), true);
+      output_permute->setAllocationDomain(
+          output_permute->getLoopDomain(), true);
```

Comment:
Setting the allocation domain for communication in/out here so that other passes cannot potentially alter that order.

Reply:
Do you agree input_permute and output_permute can simply be `set` instead of `set.permute`? 

(This may be on your TODO list for #4381 already. If so, please add a TODO.)

Reply:
This is for logical DID split. So I think it needs to be permute.

Reply:
I don't see why "needs". Can you show me an example why it's wrong to `set` with a permuted allocation domain for logical DID split?